core-wih/src/main/java/io/elimu/a2d2/corewih/MixPanelWorkItemHandler.java

```diff
@@ -0,0 +1,56 @@
+package io.elimu.a2d2.corewih;
+
+import java.io.IOException;
+
+import org.json.JSONObject;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemHandler;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mixpanel.mixpanelapi.MessageBuilder;
+import com.mixpanel.mixpanelapi.MixpanelAPI;
+
+public class MixPanelWorkItemHandler implements WorkItemHandler {
+
+	private static final Logger LOG = LoggerFactory.getLogger(MixPanelWorkItemHandler.class);
+	public static final String MIXPANEL_PREFIX = "mixpanel_";
+	
+	@Override
+	public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
+		String appToken = System.getProperty("mixpanel.app.token");
+		String distinctId = (String) workItem.getParameter("distinctId");
+		if (appToken == null || distinctId == null) {
+			LOG.info("No mixpannel app token or distinct ID provided. Returning without sending event");
+			manager.completeWorkItem(workItem.getId(), workItem.getResults());
+			return;
+		}
+		String evtName = (String) workItem.getParameter("eventDescription");
+		if (evtName == null || "".equals(evtName.trim())) {
+			evtName = "Omnibus Generic Event";
```

Comment:
I would add this to the list of "required parameters" above and return an error instead of using this default event name.